src/openrct2-ui/windows/Finances.cpp

```diff
@@ -482,15 +482,23 @@ class FinancesWindow final : public Window
             case WIDX_LOAN_INCREASE:
             {
                 auto newLoan = gBankLoan + 1000.00_GBP;
+                if (gBankLoan < gMaxBankLoan)
+                {
+                    newLoan = std::min(gMaxBankLoan, newLoan);
+                }
                 auto gameAction = ParkSetLoanAction(newLoan);
                 GameActions::Execute(&gameAction);
                 break;
             }
             case WIDX_LOAN_DECREASE:
             {
-                if (gBankLoan > 0)
+                if (gBankLoan != 0) // prevent error message if loan is exactly 0
                 {
                     auto newLoan = gBankLoan - 1000.00_GBP;
+                    if (gBankLoan > 0)
+                    {
+                        newLoan = std::max(0LL, newLoan);
+                    }
```

Comment:
Isn't this just:
```suggestion
                    auto newLoan = std::max(0.00_GBP, gBankLoan - 1000.00_GBP);
```

Reply:
Similar as above, but not exactly the same, as there is no error message involved in this case.
Here, the important part is that pressing the decrease button while the loan is negative would set the loan to zero in your solution.